megablocks/layers/moe.py

```diff
@@ -374,6 +377,7 @@ def parallel_forward_once(self, x, expert_weights, top_experts):
 
         # Locally permute the tokens and perform the expert computation.
         # Block to make sure that the cross-device permutation is complete.
+        parallel_tokens_per_expert = parallel_tokens_per_expert_cpu if isinstance(self.mlp, mlp.GroupedMLP) else parallel_tokens_per_expert
```

Comment:
Can we add a NOTE explaining what this is doing and why? Also, maybe this line is longer than 80 characters as is?

Reply:
Forgot to push last PR which simplifies this